app/services/course/user_registration_service.rb

```diff
@@ -5,16 +5,30 @@ class Course::UserRegistrationService
   # @return [bool] True if the registration succeeded. False if the registration failed.
   def register(registration)
     CourseUser.transaction do
-      if registration.code.empty?
-        register_course_user(registration)
+      course_user = create_or_update_registration(registration)
+      succeeded = course_user && !course_user.changed?
+      if succeeded && course_user.requested?
+        notify_course_staff(registration.course, course_user)
       else
-        claim_registration_code(registration)
+        succeeded
```

Comment:
Should it be:
```ruby
if succeeded && course_user.requested?
  notify_course_staff(registration.course, course_user)
end

succeeded
```

because `notify_course_staff` does not return any value.